mne/preprocessing/ica.py

```diff
@@ -3173,15 +3189,17 @@ def _find_max_corrs(all_maps, target, threshold):
 
     if len(maxmaps) == 0:
         return [], 0, 0, []
-    newtarget = np.zeros(maxmaps[0].size)
-    std_of_maps = np.std(np.asarray(maxmaps))
-    mean_of_maps = np.std(np.asarray(maxmaps))
-    for maxmap, polarity in zip(maxmaps, polarities):
-        newtarget += (maxmap / std_of_maps - mean_of_maps) * polarity
-
-    newtarget /= len(maxmaps)
-    newtarget *= std_of_maps
+    # > Next, an average cluster map is calculated, after inversion of those
+    # > ICs showing a negative correlation (sign ambiguity problem) and root
+    # > mean square (RMS) normalization of each individual IC.
+    #
+    # Which is this (rms=Frobenius norm=np.linalg.norm):
+    newtarget = np.mean(
+        [(m * p) / np.linalg.norm(m) for m, p in zip(maxmaps, polarities)], axis=0
+    )
```

Comment:
Can't this be written in vectorized form that would also be more readable? Like

```Python
(np.array(maxmaps) * np.array(polarities) / np.linalg.norm(np.array(maxmaps))).mean()
```

or something like that.

Reply:
I think that might work if you pass the right `axis` (and maybe `keepdims=True`) to `np.linalg.norm`

Reply:
Maybe with the right axis manipulations... I'll try it and you can see if it's better

Reply:
... actually I'm not sure this will work unless we're guaranteed non-raggedness of the `np.array`, which we might not be if there are different numbers of components (which I think could happen if the number of PCA vectors differs across subjects for example). But I can unify the iteration step at least to a single one, which I'll push.

Reply:
yeah we cannot assume non-raggedness (IIRC I think raggedness even happens in our tutorial docs on corrmap)

Reply:
In that case I think what's here is about as readable as you can get with as few temporaries in memory (just one for the current `sum` plus one that is being added)